### DISCUSSION TITLE
node-red-contrib-sonos-plus 6.4.1 fails to install

### DISCUSSION THREAD

**Discussion by mirodd1981** [hklages/node-red-contrib-sonos-plus#234] [Q&A] [answered]
Hello,

i'm using RaspberryMatic on a rpi3:  

> "VERSION": "3.65.11.20221005",
      "PRODUCT": "raspmatic_rpi3",
      "PLATFORM": "rpi3-aarch64",

with version of nodered:

> "redmatic": "7.2.1",
   "nodejs": "14.16.0",
   "@node-red-contrib-themes/midnight-red": "1.4.7",
   "ain2": "3.0.0",
   "node-red": "1.2.9",
   "npm": "7.6.3",
   "redmatic-homekit": "3.3.0",
   "node-red-contrib-amazon-echo": "0.1.10",
   "node-red-contrib-better-sonos": "0.1.21",
   "node-red-contrib-ccu": "3.4.2",
   "node-red-contrib-combine": "1.8.1",
   "node-red-contrib-sonos-events": "1.1.0",
   "node-red-contrib-sonos-plus": "6.4.1",
   "node-red-dashboard": "3.2.3",
   "node-red-node-rbe": "0.5.0",
   "node-red-node-tail": "0.3.2",
   "redmatic-led": "1.1.0",
   "redmatic-webapp": "0.3.1"

i tried to install node-red-contrib-sonsos-plus, but i did not managed it. I tried several times and for the first moment, it looks like as there is an interoperability with node-red-contrib-homekit-bridged, because i was able to install it, when node-red-contrib-homekit-bridged was not installed. Vice versa, i was not able to install homekit while sonos plus was installed. So i changed my flows and removed the homekit node. 
But now, i didn´t even get sonos back to work. In the meanwhile i have completely deinstalled nodered and reinstalled everything. Still not working and i get allways the same error message:

> Dec` 11 12:19:29 ccu3 daemon.debug node-red[3857]: [out] 
Dec 11 12:19:29 ccu3 daemon.debug node-red[3857]: [out] up to date in 6s
Dec 11 12:19:29 ccu3 daemon.debug node-red[3857]: rc=0
Dec 11 12:19:29 ccu3 daemon.info node-red[3857]: Installed module: node-red-contrib-sonos-plus
Dec 11 12:19:29 ccu3 daemon.warn node-red[3857]: [node-red-contrib-sonos-plus] Node module cannot be loaded on this version. Requires: >=2.0.0 

What`s the problem about? i would appreciate some help. Thanks in advance.

**Comment by mirodd1981** [comment]
I think i managed it: node-red must be version >=2.0.0. and I have 1.2.9.

As i don't want to update node-red, the question is: what's the latest version i could use?

**Comment by hklages** [comment, marked as answer]
Hi. 
I changed that because of the "scorecard": [see](https://flows.nodered.org/node/node-red-contrib-sonos-plus/scorecard#)
It is recommended to have at least 2.2.3 or even better 3. 

Redmatic seems to be the inhibitor for upgrading - right?

I guess sonos-plus 6.0.0 should work.

**Reply by mirodd1981** [reply]
Thanks a lot, and yes, Redmatic is the problem. 
I assume that i need to migrate to a dedicated server or raspberry, as it looks like that redmatic is not maintained any more. Anyway, there are some hints that an update to version 2 or 3 is possible, but i don`t want to risk a broken redmatic.

so at least: version 6.0.0 works well and I used your recommendation for the downgrade:

> npm` install --no-package-lock --global-style --save-prefix="~" --only=prod node-red-contrib-sonos-plus@6.0.0

**Comment by hklages** [comment]
Thanks for you feedback - good to hear that it works now. 

Initially this package was written for Redmatic (on a CCU3) :-)

Last year I replaced the ccu3 by an raspberry pi 4 with home assistant, virtual raspberrymatic and node-red as add on. The migration took some time but now I benefit from the hundreds of integrations such as zigbee, nas, ...
